SPT 4.0 Support + optional FIKA support

Beat2er opened the pull request:
Hi @m-barneto, 
I updated your wonderful mod to SPT 4.0 compatibility and added (optional) support for FIKA.
Pleas let me know, if the code changes align with your vision and hopefully find their way into your repo. In the meantime, can we allow palyers to use this version with it's assets for an early SPT 4 build? Of course all credits remain with you as the original author.

This is purely to help the community access your great work while we wait for your official update. The MIT license allows this, but out of respect for your work and to comply with the SPT content guidelines, I wanted to ask.
Please let me know if this is acceptable to you.
Thanks again for all your contributions to the SPT modding scene!

See the asset at https://github.com/Beat2er/BossNotifier/releases/tag/2.0

Commit:
SPT 4.0 Support
Fika Support (optional)

m-barneto commented:
Nice work, thank you! I'll merge these and push out an update when I get the chance to test. Only thing I see is that you may not want to include the intel center level in the shared packet. That way clients with upgraded hideouts can still use the mod.

Great work man. I appreciate you updating the build script as well!

ChouUn commented:
Great job!
It's better to check the work status without FIKA.
I didn't have a FIKA, so I got:
```
Could not load file or assembly 'Fika.Core, Version=2.0.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
Could not load file or assembly 'Fika.Core, Version=2.0.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
Could not load file or assembly 'Fika.Core, Version=2.0.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
Could not load type of field 'BossNotifier.FikaIntegration+<>O:<0>__OnNetworkManagerCreated' (0) due to: Could not load file or assembly 'Fika.Core, Version=2.0.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
  at (wrapper managed-to-native) System.Reflection.Assembly.GetTypes(System.Reflection.Assembly,bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at GClass3670.method_1[T] () [0x00015] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at GClass3670.method_0 () [0x00016] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at GClass3670..ctor () [0x00048] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at GClass3670.get_Instance () [0x00009] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at GClass3670.CreateData[T] (EFT.DataProviding.EDataLifeTime dataLifeTime) [0x00000] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at GlobalEventHandlerClass.Initialize () [0x00009] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at EFT.AbstractApplication.Awake () [0x00110] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at EFT.ClientApplication`1[T].Awake () [0x00000] in <ad47400a25db462cbf0a531df654c3c3>:0 
  at EFT.CommonClientApplication`1[T].Awake () [0x00045] in <ad47400a25db462cbf0a531df654c3c3>:0 
```
Maybe the FIKA support should be loaded dynamically.

Beat2er commented:
I'll take a look at both these concerns. (and yes, fika should be loaded dynamically 😇)

Commit:
Remove intel center level from Fika packets

m-barneto commented:
[https://github.com/m-barneto/BossNotifier/blob/server-component/Client/Plugin.cs](https://github.com/m-barneto/BossNotifier/blob/server-component/Client/Plugin.cs)

Has some info regarding a soft dependency on Fika. Although I had to use reflection for my fika methods doing it this way.

Commit:
Implement two-DLL Fika integration

Split Fika integration into separate BossNotifier.FikaOptional.dll.bin assembly to support optional Fika. Main plugin loads .bin file conditionally when Fika is detected, allowing mod to work in both singleplayer and multiplayer modes without errors.

This removes the need for complex reflection throughout the Fika integration code - FikaIntegration.cs uses direct references while remaining optional.

Commit:
Bump version to 2.0.1

Beat2er commented:
@m-barneto take a look at this approach. This seems to be much more easy to handle/maintain without so much complex reflection. Also this seems like a good approach of separation of services/features. Let me know, what you think about it. :)

Fixed:
- client's intel center
- move fika to optional assembly (which automatically gets loaded by main one if Fika is present)

Let me know if you have any concerns or find any problems testing (also directed at players, not especially m-barneto; I don't want to pile so much work on you, but rather take some off your plate. Just everyone who wants to test/use the early version)
https://github.com/Beat2er/BossNotifier/releases/tag/2.0.1

Beat2er commented:
Hi @m-barneto, did you have any time to look at this? Or may we temporarly link to a prerelease version hosted on my github until you find time to take over development again?
